### PR TITLE
Avoid reloading file icons in the Assets Library #3211

### PR DESCRIPTION
**Pull request by suppermancool**
Avoid reloading file icons in the Assets Library #3211

### CODE REVIEW

**Comment by suppermancool** [comment]
@maxceem actually all images are cached by browser i'm not sure why it didn't work on you pc. I will try implement the way `project type icons` does (it loaded all icon into code, this is not the good way i think).
I think this error will happen for all the image in the app and not just the icon in Assets Library and i don't think we can implement the way of `project type icons` for all the image.

In my site the browser work good and it cached all loaded image:
https://drive.google.com/file/d/11_zByr_hQlLdPa8GL7et1CPbojM60Ebm/view?usp=sharing

**Comment by maxceem** [comment]
> I think this error will happen for all the image in the app and not just the icon in Assets Library and i don't think we can implement the way of project type icons for all the image.

Yeah, agree, though I think it's ok for the images which are "content" to be loaded during page changing. But the images which are the part of UI is better to keep always loaded so the whole UI is shown instantly.

**Comment by suppermancool** [comment]
ah actually this should be the way reactjs does, it alway import the image before using it, i almost forget this 🤦‍♂️

**Comment by suppermancool** [comment]
@maxceem done in the latest commit

File: src/components/FileIcon.jsx [hunk 1]
```diff
@@ -0,0 +1,230 @@
+/**
+ * FileIcon component
+ *
+ * Renders proper project type icon depend on the project type
```
Comment (maxceem) [inline src/components/FileIcon.jsx]: comment left from another component

Comment (suppermancool) [reply src/components/FileIcon.jsx]: @maxceem all done in the latest commit

**Pull request merged**